tiledb/sm/query/query.cc

```diff
@@ -2041,25 +2041,59 @@ tdb_shared_ptr<Buffer> Query::rest_scratch() const {
 }
 
 bool Query::use_refactored_dense_reader() {
+  bool use_refactored_readers = false;
   bool found = false;
-  std::string val = config_.get("sm.query.dense.reader", &found);
+  // First check for legacy option
+  config_.get<bool>(
+      "sm.use_refactored_readers", &use_refactored_readers, &found);
+  // If the legacy option is set use it over the new parameters
+  // This facilitates backwards compatibility
+  if (found)
+    return use_refactored_readers;
+
+  config_.get<bool>("sm.query.dense.reader", &use_refactored_readers, &found);
   assert(found);
-  return val.compare("refactored") == 0;
+
+  return use_refactored_readers;
```

Comment:
Why don't we use the new readers as default for >= 2.5?

Reply:
We use the sparse unordered reader by default now but sparse global order and dense are still defaulting to legacy. We expect to move over to those in the near future.

Reply:
Thanks